Task/Task-2/doubleLinkedList.cpp

```diff
@@ -46,35 +46,27 @@ template <typename T> size_t DoubleLinkedList<T>::length() const {
 }
 
 template <typename T> void DoubleLinkedList<T>::push_front(const T &data) {
-  Node *node = new Node(data);
-  if (empty()) {
-    head = node;
-    tail = node;
-    head->prev = tail;
-    tail->next = head;
+  std::unique_ptr<Node> new_node = std::make_unique<Node>(data);
+  if (head == nullptr) {
+    head = std::move(new_node);
+    tail = head.get();
   } else {
-    node->next = head;
-    head->prev = node;
-    head = node;
-    head->prev = tail;
-    tail->next = head;
+    new_node = std::move(head);
+    head = std::move(new_node);
+    head->next->prev = head.get();
   }
   size++;
 }
 
 template <typename T> void DoubleLinkedList<T>::push_back(const T &data) {
-  Node *node = new Node(data);
-  if (empty()) {
-    head = node;
-    tail = node;
-    head->prev = tail;
-    tail->next = head;
+  std::unique_ptr<Node> new_node = std::make_unique<Node>(data);
```

Comment:
https://stackoverflow.com/questions/47892683/smart-pointers-in-doubly-linked-list